matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/media/MediaService.kt

```diff
@@ -17,15 +17,17 @@
 package org.matrix.android.sdk.api.session.media
 
 import org.matrix.android.sdk.api.cache.CacheStrategy
-import org.matrix.android.sdk.api.session.events.model.Event
+import org.matrix.android.sdk.api.session.room.timeline.TimelineEvent
 import org.matrix.android.sdk.api.util.JsonDict
 
 interface MediaService {
     /**
-     * Extract URLs from an Event.
-     * @return the list of URLs contains in the body of the Event. It does not mean that URLs in this list have UrlPreview data
+     * Extract URLs from a TimelineEvent.
+     * @param event TimelineEvent to extract the URL from.
+     * @param forceExtract Should be used for edited events. If true, URL will be extracted again even it is already in the cache.
+     * @return the list of URLs contains in the body of the TimelineEvent. It does not mean that URLs in this list have UrlPreview data
      */
-    fun extractUrls(event: Event): List<String>
+    fun extractUrls(event: TimelineEvent, forceExtract: Boolean = false): List<String>
```

Comment:
It should be possible to avoid having the parameter `forceExtract` and let the SDK have a better management of the cache, maybe by using the latest eventId which edit the original message as a key, and not the eventId itself.

Reply:
I did it